src/gc.c

```diff
@@ -818,9 +818,10 @@ FORCE_INLINE int gc_try_setmark_tag(jl_taggedvalue_t *o, uint8_t mark_mode) JL_N
         tag = tag | mark_mode;
         assert((tag & 0x3) == mark_mode);
     }
-    tag = jl_atomic_exchange_relaxed((_Atomic(uintptr_t)*)&o->header, tag);
+    uintptr_t curr = jl_atomic_load_relaxed((_Atomic(uintptr_t)*)&o->header);
```

Comment:
Should we make the load in 807 also relaxed to match here?

Also as we discussed on slack not sure that the load here is super necessary (can you disassemble this function?) / Wouldn't need a compiler barrier.

Reply:
By this you mean making the load above atomic_relaxed and just reusing it? That's fine by me.